### PR TITLE
Spack package manger added

### PR DESCRIPTION
**Pull request by JensWehner**
- [x] I followed the [CONTRIBUTING guidelines](../blob/master/CONTRIBUTING.md).

Below, describe what this Pull Request adds:

Mentioned spack package manager.

### CODE REVIEW

File: best_practices/language_guides/ccpp.md [hunk 1]
```diff
@@ -133,6 +133,9 @@ With guix, you can create virtual environments, much like those in Python `virtu
 You can also create relocatable binaries to use your binaries on systems that do not have guix installed.
 This makes it easy to test your packages on your laptop before deploying to a cluster system.
 
+A package that gains more traction at the moment for HPC environments is [spack](https://spack.readthedocs.io/en/latest/).
+The main advantage of Spack over Conda is its ability to choose a specific compiler and BLAS/LAPACK or MPI library. Spack also has better platform support for supercomputers. On the other hand, Conda offers Windows support.
```
Comment (egpbos) [inline best_practices/language_guides/ccpp.md:137]: I'm not sure this advantage is true. In principle, with Conda, you can build and install any kind of package you like as well. Maybe a better way to put it is something like: "Spack allows you to pick from many compilers. When installing packages, it compiles every package from scratch. This allows you to be tailor compilation flags and such to take fullest advantage of your cluster's hardware, which can be essential in HPC situations." I would not mention Conda at all, I think this way it's quite clear what the advantage of Spack is.

What do you mean exactly with "better platform support"?

**Comment by egpbos** [comment]
Nice addition! Some comments above. Just out of curiosity, did you use Spack in projects?

**Comment by JensWehner** [comment]
Only as a user.

File: best_practices/language_guides/ccpp.md (hunk 1 above)
Comment (JensWehner) [reply best_practices/language_guides/ccpp.md]: this is actually much clearer. Thanks

**Comment by egpbos** [comment]
Also here, build errors were caused by different issue, addressed previously.

**Pull request merged**